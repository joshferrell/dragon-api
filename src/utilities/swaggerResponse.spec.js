import * as swaggerFormat from './swaggerResponse';

describe('swagger formatting', () => {
    it('should have a format for success', () => {
        expect(swaggerFormat.successFormat).toMatchSnapshot();
    });

    it('should have a format for bad requests', () => {
        expect(swaggerFormat.badRequestFormat).toMatchSnapshot();
    });

    it('should have a format for unauthorized requests', () => {
        expect(swaggerFormat.unauthorizedFormat).toMatchSnapshot();
    });

    it('should have an internal error format', () => {
        expect(swaggerFormat.internalErrorFormat).toMatchSnapshot();
    });

    it('should have a not implemented format', () => {
        expect(swaggerFormat.notImplementedFormat).toMatchSnapshot();
    });
});
