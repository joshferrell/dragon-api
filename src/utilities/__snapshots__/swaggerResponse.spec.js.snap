// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`swagger formatting should have a format for bad requests 1`] = `
Object {
  "400": Object {
    "description": "Bad Request",
    "schema":  {
      "_baseType": undefined,
      "_description": null,
      "_examples": Array [],
      "_flags": Object {
        "presence": "required",
      },
      "_inner": Object {
        "children": Array [
          Object {
            "key": "statusCode",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "400",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "error",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "Bad Request",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "message",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "validation",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {},
              "_inner": Object {
                "children": Array [
                  Object {
                    "key": "source",
                    "schema":  {
                      "_baseType": undefined,
                      "_description": null,
                      "_examples": Array [],
                      "_flags": Object {
                        "presence": "required",
                      },
                      "_inner": Object {},
                      "_invalids": Set {
                        "_set": Array [
                          "",
                        ],
                      },
                      "_meta": Array [],
                      "_notes": Array [],
                      "_refs": Array [],
                      "_settings": null,
                      "_tags": Array [],
                      "_tests": Array [],
                      "_type": "string",
                      "_unit": null,
                      "_valids": Set {
                        "_set": Array [],
                      },
                      "isJoi": true,
                    },
                  },
                  Object {
                    "key": "keys",
                    "schema":  {
                      "_baseType": undefined,
                      "_description": null,
                      "_examples": Array [],
                      "_flags": Object {
                        "presence": "required",
                        "sparse": false,
                      },
                      "_inner": Object {
                        "exclusions": Array [],
                        "inclusions": Array [
                           {
                            "_description": null,
                            "_examples": Array [],
                            "_flags": Object {},
                            "_inner": Object {},
                            "_invalids": Set {
                              "_set": Array [
                                "",
                              ],
                            },
                            "_meta": Array [],
                            "_notes": Array [],
                            "_refs": Array [],
                            "_settings": null,
                            "_tags": Array [],
                            "_tests": Array [],
                            "_type": "string",
                            "_unit": null,
                            "_valids": Set {
                              "_set": Array [],
                            },
                            "isJoi": true,
                          },
                        ],
                        "items": Array [
                           {
                            "_description": null,
                            "_examples": Array [],
                            "_flags": Object {},
                            "_inner": Object {},
                            "_invalids": Set {
                              "_set": Array [
                                "",
                              ],
                            },
                            "_meta": Array [],
                            "_notes": Array [],
                            "_refs": Array [],
                            "_settings": null,
                            "_tags": Array [],
                            "_tests": Array [],
                            "_type": "string",
                            "_unit": null,
                            "_valids": Set {
                              "_set": Array [],
                            },
                            "isJoi": true,
                          },
                        ],
                        "ordereds": Array [],
                        "requireds": Array [],
                      },
                      "_invalids": Set {
                        "_set": Array [],
                      },
                      "_meta": Array [],
                      "_notes": Array [],
                      "_refs": Array [],
                      "_settings": null,
                      "_tags": Array [],
                      "_tests": Array [],
                      "_type": "array",
                      "_unit": null,
                      "_valids": Set {
                        "_set": Array [],
                      },
                      "isJoi": true,
                    },
                  },
                ],
                "dependencies": Array [],
                "patterns": Array [],
                "renames": Array [],
              },
              "_invalids": Set {
                "_set": Array [],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "object",
              "_unit": null,
              "_valids": Set {
                "_set": Array [],
              },
              "isJoi": true,
            },
          },
        ],
        "dependencies": Array [],
        "patterns": Array [],
        "renames": Array [],
      },
      "_invalids": Set {
        "_set": Array [],
      },
      "_meta": Array [],
      "_notes": Array [],
      "_refs": Array [],
      "_settings": null,
      "_tags": Array [],
      "_tests": Array [],
      "_type": "object",
      "_unit": null,
      "_valids": Set {
        "_set": Array [],
      },
      "isJoi": true,
    },
  },
}
`;

exports[`swagger formatting should have a format for success 1`] = `
Object {
  "200": Object {
    "description": "Success",
    "schema":  {
      "_baseType": undefined,
      "_description": null,
      "_examples": Array [],
      "_flags": Object {
        "presence": "required",
      },
      "_inner": Object {
        "children": Array [
          Object {
            "key": "success",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "insensitive": true,
                "presence": "required",
              },
              "_inner": Object {
                "falsySet": Set {
                  "_set": Array [],
                },
                "truthySet": Set {
                  "_set": Array [],
                },
              },
              "_invalids": Set {
                "_set": Array [],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "boolean",
              "_unit": null,
              "_valids": Set {
                "_set": Array [],
              },
              "isJoi": true,
            },
          },
        ],
        "dependencies": Array [],
        "patterns": Array [],
        "renames": Array [],
      },
      "_invalids": Set {
        "_set": Array [],
      },
      "_meta": Array [],
      "_notes": Array [],
      "_refs": Array [],
      "_settings": null,
      "_tags": Array [],
      "_tests": Array [],
      "_type": "object",
      "_unit": null,
      "_valids": Set {
        "_set": Array [],
      },
      "isJoi": true,
    },
  },
}
`;

exports[`swagger formatting should have a format for unauthorized requests 1`] = `
Object {
  "401": Object {
    "description": "Unauthorized",
    "schema":  {
      "_baseType": undefined,
      "_description": null,
      "_examples": Array [],
      "_flags": Object {
        "presence": "required",
      },
      "_inner": Object {
        "children": Array [
          Object {
            "key": "statusCode",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "401",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "error",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "Unauthorized",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "message",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [],
              },
              "isJoi": true,
            },
          },
        ],
        "dependencies": Array [],
        "patterns": Array [],
        "renames": Array [],
      },
      "_invalids": Set {
        "_set": Array [],
      },
      "_meta": Array [],
      "_notes": Array [],
      "_refs": Array [],
      "_settings": null,
      "_tags": Array [],
      "_tests": Array [],
      "_type": "object",
      "_unit": null,
      "_valids": Set {
        "_set": Array [],
      },
      "isJoi": true,
    },
  },
}
`;

exports[`swagger formatting should have a not implemented format 1`] = `
Object {
  "501": Object {
    "description": "Not Implemented",
    "schema":  {
      "_baseType": undefined,
      "_description": null,
      "_examples": Array [],
      "_flags": Object {
        "presence": "required",
      },
      "_inner": Object {
        "children": Array [
          Object {
            "key": "statusCode",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "501",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "error",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "Not Implemented",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "message",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "method not implemented",
                ],
              },
              "isJoi": true,
            },
          },
        ],
        "dependencies": Array [],
        "patterns": Array [],
        "renames": Array [],
      },
      "_invalids": Set {
        "_set": Array [],
      },
      "_meta": Array [],
      "_notes": Array [],
      "_refs": Array [],
      "_settings": null,
      "_tags": Array [],
      "_tests": Array [],
      "_type": "object",
      "_unit": null,
      "_valids": Set {
        "_set": Array [],
      },
      "isJoi": true,
    },
  },
}
`;

exports[`swagger formatting should have an internal error format 1`] = `
Object {
  "500": Object {
    "description": "Internal Server Error",
    "schema":  {
      "_baseType": undefined,
      "_description": null,
      "_examples": Array [],
      "_flags": Object {
        "presence": "required",
      },
      "_inner": Object {
        "children": Array [
          Object {
            "key": "statusCode",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "500",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "error",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "Internal Server Error",
                ],
              },
              "isJoi": true,
            },
          },
          Object {
            "key": "message",
            "schema":  {
              "_baseType": undefined,
              "_description": null,
              "_examples": Array [],
              "_flags": Object {
                "allowOnly": true,
                "presence": "required",
              },
              "_inner": Object {},
              "_invalids": Set {
                "_set": Array [
                  "",
                ],
              },
              "_meta": Array [],
              "_notes": Array [],
              "_refs": Array [],
              "_settings": null,
              "_tags": Array [],
              "_tests": Array [],
              "_type": "string",
              "_unit": null,
              "_valids": Set {
                "_set": Array [
                  "An internal server error occurred",
                ],
              },
              "isJoi": true,
            },
          },
        ],
        "dependencies": Array [],
        "patterns": Array [],
        "renames": Array [],
      },
      "_invalids": Set {
        "_set": Array [],
      },
      "_meta": Array [],
      "_notes": Array [],
      "_refs": Array [],
      "_settings": null,
      "_tags": Array [],
      "_tests": Array [],
      "_type": "object",
      "_unit": null,
      "_valids": Set {
        "_set": Array [],
      },
      "isJoi": true,
    },
  },
}
`;
